### PR TITLE
feat: add TSConfigInfo to display current config

### PR DESCRIPTION
**Pull request by theHamsta**
https://nvim-treesitter.zulipchat.com/#narrow/stream/252271-general/topic/Random/near/210929394

### CODE REVIEW

File: lua/nvim-treesitter/configs.lua [hunk 1]
```diff
@@ -222,6 +222,15 @@ local function recurse_modules(accumulator, root, path)
...
 end
 
+function M.config_info(process_function)
+  process_function = process_function or function(item, path)
+     if path[#path] == vim.inspect.METATABLE then return end
+     if path[#path] == "is_supported" then return end
+     return item
+  end
+  print(vim.inspect(config, {process = process_function}))
+end
```
Comment (theHamsta) [inline lua/nvim-treesitter/configs.lua:232]: could also make this a local function

Comment (vigoux) [reply lua/nvim-treesitter/configs.lua:235]: I'd rather have in as a local function.

**Comment by kyazdani42** [comment]
i was wondering if using a buffer to display stats about the plugin would be better.

**Comment by vigoux** [comment]
> i was wondering if using a buffer to display stats about the plugin would be better.

I don't think so, just having everything dumped in the message window is fine IMO.

**Comment by theHamsta** [comment]
I considered using a read-only floating window..

**Comment by kyazdani42** [comment]
dumping everything is just not working for me on small screens, we cannot see anything at all, it's all messed up, that's why i was suggesting otherwise.

**Comment by kyazdani42** [comment]
take the vim plug way of displaying status, it just works great for me on any screen :)

**Review by kyazdani42 — APPROVED** [review]

**Review by vigoux — APPROVED** [review]

**Pull request merged**